framework/src/st_things/things_stack/utils/things_network.c

```diff
@@ -380,12 +381,41 @@ void things_wifi_scan_done(wifi_manager_scan_info_s **scan_result, int res)
 	access_point_info_s *pinfo = NULL;
 	access_point_info_s *p_last_info = NULL;
 	while (wifi_scan_iter != NULL) {
-		if (wifi_scan_iter->ssid != NULL) {
+		if (strlen(wifi_scan_iter->ssid) != 0) {
```

Comment:
I don't agree with your opinion.
If the strlen argument is null, a segmentation error will occur.
And do you need checks other than ssid? (eq. bssid, rssi, ...)

Reply:
Now I changed it to :-
if ((wifi_scan_iter->ssid != NULL) && (strlen(wifi_scan_iter->ssid)) != 0)

Reply:
I suggested to change "if (wifi_scan_iter->ssid != NULL)" to "if (wifi_scan_iter->ssid != NULL && strlen(wifi_scan_iter->ssid) != 0)" to avoid segmentation fault,

May i know why you don't agree with this opinion?